src/electionguard/api/encrypt_ballot.c

```diff
@@ -113,7 +129,113 @@ void API_EncryptBallot_free(struct register_ballot_message message,
     }
 
     if (tracker_string != NULL)
+    {
         free(tracker_string);
+        tracker_string = NULL;
+    }
+}
+
+bool API_EncryptBallot_soft_delete_file(char *export_path, char *filename)
+{
+    bool ok = true;
+    char *default_prefix = "electionguard_encrypted_ballots-";
+    char *existing_filename = malloc(FILENAME_MAX + 1);
+    if (existing_filename == NULL)
+    {
+        ok = false;
+        return ok;
+    }
+
+    char *soft_delete_filename = malloc(FILENAME_MAX + 1);
+    if (soft_delete_filename == NULL)
+    {
+        ok = false;
+        return ok;
+    }
+
+    ok = generate_filename(export_path, filename, default_prefix, existing_filename);
+
+    ok = generate_unique_filename(export_path, filename, default_prefix, soft_delete_filename);
+
+    uint32_t result = rename(existing_filename, soft_delete_filename);
+
+#ifdef DEBUG_PRINT 	
+   if(result != 0) 
+   {
+      printf("API_EncryptBallot_soft_delete_file: unable to rename the file\n\n");
+   }
+#endif
+
+    free(existing_filename);
+    free(soft_delete_filename);
+
+    return ok;
+}
+
+bool export_ballot(char *export_path, char *filename, char **output_filename, 
+                    char *identifier,
+                       struct register_ballot_message *encrypted_ballot_message)
+{
+    bool ok = true;
+    char *default_prefix = "electionguard_encrypted_ballots-";
+    *output_filename = malloc(FILENAME_MAX + 1);
+    if (output_filename == NULL)
+    {
+        ok = false;
+        return ok;
+    }
+    
+    ok = generate_filename(export_path, filename, default_prefix, *output_filename);
+#ifdef DEBUG_PRINT 
+    printf("API_EncryptBallots: generated filename for export at \"%s\"\n", *output_filename); 
+#endif
+
+    if (ok && !Directory_exists(export_path))
+    {
+        ok = create_directory(export_path);
+    } 
+
+    if (ok)
+    {
+        FILE *out = fopen(*output_filename, "a+");
+        if (out == NULL)
+        {
+            printf("API_EncryptBallots: error accessing file\n");
+            return false;
+        }
+
+        int seek_status = fseek(out, 0, SEEK_END);
+        if (seek_status != 0)
+        {
+            printf("API_EncryptBallots: error seeking file\n");
+            return false;
+        }
+        
+        enum Voting_Encrypter_status status =
+            Voting_Encrypter_write_ballot(out, identifier, encrypted_ballot_message);
```

Comment:
Not clear if `status` value is lost (local variable), or if a copy is encoded in `encrypted_ballot_message`.

Reply:
I think this raises a larger architectural question.  Internally the library is using enums to define return states which could have state-change side effects, but from the api's perspective we are trying to move purely to success/fail (hence the bool return values) and make the `API_` prefixed method calls idempotent.

one reason is to simplify the api surface for callers that wrap this lib in other languages to reduce the likelihood of having to make code changes to dependent libraries due to changes in return status codes, etc.  Another was to provide essentially two ways of interacting with the library: 
- the `API_` level that is simplified covering the ~80% use cases
- the `XXX_Coordinator_` level that is more advanced for consumers who will customize their wrappers anyway.

perhaps a better approach would be to return a struct indicating success/fail and a string message with a description?  My concern is that the enums may not be consistently applied throughout the library and returning integer-based codes may not have the same meaning across all components as the library evolves.  Another option would be to implement some consistent error handling by implementing a global exception pattern that all of the coordinators can follow, but i think that is probably out of scope of this PR.

In the short term, i lean towards logging this as a feature enhancement and calling out this specific line of code as an example of where this needs to be fixed.  That way we can scope this properly as part of another milestone.  

What do you think?